.github/workflows/dispatch-release-pypi.yml

```diff
@@ -31,6 +32,10 @@ jobs:
           python -m pip install --upgrade pip
           pip install setuptools wheel twine
 
+      - name: Set Library Version as ${{ github.ref_name }}
+        run: |
+          echo "__version__ = \"${{ github.ref_name }}\"" > learnosity_sdk/_version.py
```

Comment:
This is a good idea, but this well create a discrepancy between the code in the repo, and the code in the release.

It would make sense to do this automatically, but it should be commited back into git, otherwise the change will only ever exist this one-time during the GitHub actions run.

Reply:
> otherwise the change will only ever exist this one-time during the GitHub actions run.
Yes that's entirely right.

My thought would be the version will always be present in the packaged version of the library, or if you're working on the code, you could tell by the most recent git tag.

This achieves the single source of truth, but may not be want we want. Definitely a trade off to think through.

Reply:
Yeah, the general objective makes sense, but I think we need a mechanism to determine the most recent tag and put it in the right place if the version file is missing. I tend to use `git describe`, but it may create old looking version files by referring to the last release, rather than the next.

Reply:
I think this may not play well with the Python versioning, and it will be a problem with the telemetry. We want to know what versions are in use so we know when we can deprecate features no longer in use.

Reply:
> It will be a problem with the telemetry

Hmm ok. Maybe a simple sanity check on the deployment workflow to check if the `__version__` matches the git tag and to fail the workflow otherwise?

That could provide the benefit of:
1. prevents accidental overwrites of the same version in PyPI
2. Ensures git tag and python version are in sync
3. Retains the `__version__` in git for Telemetry?

Reply:
Something like this? https://github.com/Learnosity/learnosity-sdk-python/blob/master/Makefile#L75-L80

Reply:
Oh perfect!, thanks for highlighting that! 🚀 

I'll add that to this workflow so we don't have to re-invent the wheel! 😆

Reply:
I tried to use the `Makefile` but found it not to be working as expected.

It's a simple one-liner in the end with `grep` and flexible with the github.ref_name from the below discussion.
https://github.com/orgs/community/discussions/26686